configure.sh

```diff
@@ -29,7 +29,8 @@ General options;
 Features:
 The following flags enable optional features (disable with --no-<option name>).
   --static                 build static libraries and binaries [default=no]
-  --static-binary          enable/disable static binaries
+  --static-binary          force using standard libraries statically
+                           (must be disabled for static macOS builds) [default=yes]
```

Comment:
I wonder if we should do some more magic in the configure.sh script to automatically macOS and disable --static-binary.

Reply:
Sounds good. If we do that -- can/should we remove `static-binary` option altogether, making it transparent to the user?

Reply:
No I'd still keep the option. But since for macOS this can't be supported we can automatically disable it on macOS.